### PR TITLE
rest/database: added patch-worker

### PR DESCRIPTION
**Pull request by LeanderK**
- added patch-worker operation to the router.

### CODE REVIEW

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/rest/resources/WorkerResource.java [hunk 1]
```diff
@@ -115,6 +115,26 @@ public Worker put(Request request, Response response) {
...
+    public Worker patch(Request request, Response response) {
+        Worker worker = request.attribute("input");
+        int id = getParamInt(request, "id");
+
+        worker = operations.updateWorker(worker, id);
+
+        EventManager.WORKER_CHANGE.emit(worker);
+
+        response.status(201);
+        response.header("Location", "/workers/" + worker.getId());
```
Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/rest/resources/WorkerResource.java]: Status code and header should be removed.

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/WorkerOperations.java [hunk 2]
```diff
@@ -186,6 +187,9 @@ public void anonymizeWorker(int id) throws IllegalArgumentException {
      */
     public Worker insertWorker(Worker toStore, String identity) {
         assertHasField(toStore, Worker.PLATFORM_FIELD_NUMBER);
+        if (toStore.getEmail() != null && !EmailValidator.getInstance(false).isValid(toStore.getEmail())) {
```
Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/WorkerOperations.java]: Default value for strings is `""` IIRC.

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/rest/resources/WorkerResource.java [hunk 3]
```diff
@@ -46,7 +46,7 @@ public Worker identify(Request request, Response response) {
             Optional<WorkerRecord> optionalRecord = manager.getWorker(platform, request.queryMap().toMap());
             if (optionalRecord.isPresent()) {
                 worker = optionalRecord.get();
-            } else if (!manager.getNeedemail(platform) || Objects.equals(request.queryParams("email"), String.valueOf(true))){
+            } else if (!manager.getNeedemail(platform) || (!request.queryParams("email").isEmpty())){
```
Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/rest/resources/WorkerResource.java:49]: Will Spark return an empty string if it's not present or will if throw a `NPE`?

Comment (LeanderK) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/rest/resources/WorkerResource.java:49]: i think i tested it, `queryParams` returns a empty `QueryMap`

File: build.gradle [hunk 4]
```diff
@@ -45,6 +45,7 @@ dependencies {
     compile 'com.google.protobuf:protobuf-java:3.+'
     compile 'commons-validator:commons-validator:1.5.0'
     compile 'com.google.protobuf:protobuf-java-util:3.+'
+    compile 'commons-validator:commons-validator:1.5.0'
```
Comment (kelunik) [inline build.gradle]: Already present two lines above.

**Comment by LeanderK** [comment]
push @kelunik

**Pull request merged**